src/state/store.ts

```diff
@@ -0,0 +1,136 @@
+import { IFieldContext, IField, FieldRegistry } from '../data';
+
+export interface ISubscriber {
+    (): void
+}
+
+export interface FieldEditingHandler {
+    (field: IField): void;
+}
+
+// createStore is a helper function for initializing
+// the store.
+export function createStore(props: IStoreProps) {
+    return new Store(props);
+}
+
+export interface IStore {
+    // subscribe pushes the subscriber into a list and is called
+    // via notify() until unsubscribed.
+    subscribe(subscriber: ISubscriber): void;
```

Comment:
ISubscriber is to not for a single change, all store changes will notify from this Subscriber. Right?